<?xml version="1.0" ?>
<robot name="human_model" xmlns:xacro="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface">
  <link name="base"/>
  <link name="hip">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0395316530564"/>
      <geometry>
        <!--<box size="0.4 0.2 ${waist_length}" />-->
        <cylinder length="0.0790633061127" radius="0.13488508899"/>
      </geometry>
    </visual>
  </link>
  <link name="waist_0"/>
  <link name="waist_1"/>
  <link name="waist">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0404696874451"/>
      <geometry>
        <!--<box size="0.4 0.2 ${torso_length}" />-->
        <cylinder length="0.0809393748903" radius="0.15488508899"/>
      </geometry>
    </visual>
  </link>
  <link name="spine">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.122505620162"/>
      <geometry>
        <!--<box size="0.4 0.2 ${torso_length}" />-->
        <cylinder length="0.245011240325" radius="0.15488508899"/>
      </geometry>
    </visual>
  </link>
  <link name="shoulder_center">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.058058555289"/>
      <geometry>
        <!--<box size="0.4 0.2 ${torso_length}" />-->
        <cylinder length="0.116117110578" radius="0.17488508899"/>
      </geometry>
    </visual>
  </link>
  <link name="neck_0"/>
  <link name="neck_1"/>
  <link name="neck">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.092380762709"/>
      <geometry>
        <!--<box size="0.4 0.2 ${torso_length}" />-->
        <cylinder length="0.184761525418" radius="0.04"/>
      </geometry>
    </visual>
  </link>
  <link name="head">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.1"/>
      </geometry>
    </visual>
  </link>
  <joint name="hip" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base"/>
    <child link="hip"/>
  </joint>
  <!-- SPINE : spherical -->
  <joint name="waist_0" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.0790633061127"/>
    <axis xyz="1 0 0"/>
    <parent link="hip"/>
    <child link="waist_0"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="waist_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="waist_0"/>
    <child link="waist_1"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="waist_2" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="waist_1"/>
    <child link="waist"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="spine" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.0809393748903"/>
    <axis xyz="0 1 0"/>
    <parent link="waist"/>
    <child link="spine"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="shoulder_center" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.245011240325"/>
    <parent link="spine"/>
    <child link="shoulder_center"/>
  </joint>
  <!-- NECK : spherical -->
  <joint name="neck_0" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.116117110578"/>
    <axis xyz="1 0 0"/>
    <parent link="shoulder_center"/>
    <child link="neck_0"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="neck_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="neck_0"/>
    <child link="neck_1"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="neck_2" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="neck_1"/>
    <child link="neck"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <!-- END NECK -->
  <joint name="head" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.184761525418"/>
    <parent link="neck"/>
    <child link="head"/>
  </joint>
  <link name="right_arm_mount"/>
  <link name="right_shoulder_0"/>
  <link name="right_shoulder_1"/>
  <link name="right_shoulder">
    <visual>
      <origin rpy="0 1.57079632679 0" xyz="0.132139945158 0 0"/>
      <geometry>
        <cylinder length="0.264279890316" radius="0.05"/>
      </geometry>
    </visual>
  </link>
  <link name="right_elbow_0"/>
  <link name="right_elbow">
    <visual>
      <origin rpy="0 1.57079632679 0" xyz="0.0996002567535 0 0"/>
      <geometry>
        <cylinder length="0.199200513507" radius="0.025"/>
      </geometry>
    </visual>
  </link>
  <link name="right_wrist_0"/>
  <link name="right_wrist">
    <visual>
      <origin rpy="0 0 0" xyz="0.05 0 0"/>
      <geometry>
        <box size="0.1 0.05 .005"/>
      </geometry>
    </visual>
  </link>
  <link name="right_hand"/>
  <!-- Joints -->
  <joint name="right_arm_mount" type="fixed">
    <origin rpy="0 0 -1.57079632679" xyz="0 -0.17488508899 0"/>
    <parent link="shoulder_center"/>
    <child link="right_arm_mount"/>
  </joint>
  <!-- SHOULDER : spherical -->
  <joint name="right_shoulder_0" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="right_arm_mount"/>
    <child link="right_shoulder_0"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="right_shoulder_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="right_shoulder_0"/>
    <child link="right_shoulder_1"/>
    <limit effort="1" lower="-1.57079632679" upper="2.35619449019" velocity="1"/>
  </joint>
  <joint name="right_shoulder_2" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="1 0 0"/>
    <parent link="right_shoulder_1"/>
    <child link="right_shoulder"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <!-- END SHOULDER -->
  <joint name="right_elbow_0" type="revolute">
    <origin rpy="0 0 0" xyz="0.264279890316 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="right_shoulder"/>
    <child link="right_elbow_0"/>
    <limit effort="1" lower="0.0" upper="3.14159265359" velocity="1"/>
  </joint>
  <joint name="right_elbow_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="1 0 0"/>
    <parent link="right_elbow_0"/>
    <child link="right_elbow"/>
    <limit effort="1" lower="0.0" upper="3.14159265359" velocity="1"/>
  </joint>
  <joint name="right_wrist_0" type="revolute">
    <origin rpy="0 0 0" xyz="0.199200513507 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="right_elbow"/>
    <child link="right_wrist_0"/>
    <limit effort="1" lower="-0.78539816339" upper="0.78539816339" velocity="1"/>
  </joint>
  <joint name="right_wrist_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="right_wrist_0"/>
    <child link="right_wrist"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="right_hand" type="fixed">
    <origin rpy="0 0 0" xyz="0.1 0 0"/>
    <parent link="right_wrist"/>
    <child link="right_hand"/>
  </joint>
  <link name="left_arm_mount"/>
  <link name="left_shoulder_0"/>
  <link name="left_shoulder_1"/>
  <link name="left_shoulder">
    <visual>
      <origin rpy="0 1.57079632679 0" xyz="0.132139945158 0 0"/>
      <geometry>
        <cylinder length="0.264279890316" radius="0.05"/>
      </geometry>
    </visual>
  </link>
  <link name="left_elbow_0"/>
  <link name="left_elbow">
    <visual>
      <origin rpy="0 1.57079632679 0" xyz="0.0996002567535 0 0"/>
      <geometry>
        <cylinder length="0.199200513507" radius="0.025"/>
      </geometry>
    </visual>
  </link>
  <link name="left_wrist_0"/>
  <link name="left_wrist">
    <visual>
      <origin rpy="0 0 0" xyz="0.05 0 0"/>
      <geometry>
        <box size="0.1 0.05 .005"/>
      </geometry>
    </visual>
  </link>
  <link name="left_hand"/>
  <!-- Joints -->
  <joint name="left_arm_mount" type="fixed">
    <origin rpy="0 0 1.57079632679" xyz="0 0.17488508899 0"/>
    <parent link="shoulder_center"/>
    <child link="left_arm_mount"/>
  </joint>
  <!-- SHOULDER : spherical -->
  <joint name="left_shoulder_0" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="left_arm_mount"/>
    <child link="left_shoulder_0"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="left_shoulder_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="left_shoulder_0"/>
    <child link="left_shoulder_1"/>
    <limit effort="1" lower="-2.35619449019" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="left_shoulder_2" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="1 0 0"/>
    <parent link="left_shoulder_1"/>
    <child link="left_shoulder"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <!-- END SHOULDER -->
  <joint name="left_elbow_0" type="revolute">
    <origin rpy="0 0 0" xyz="0.264279890316 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="left_shoulder"/>
    <child link="left_elbow_0"/>
    <limit effort="1" lower="-3.14159265359" upper="0.0" velocity="1"/>
  </joint>
  <joint name="left_elbow_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="1 0 0"/>
    <parent link="left_elbow_0"/>
    <child link="left_elbow"/>
    <limit effort="1" lower="-3.14159265359" upper="0.0" velocity="1"/>
  </joint>
  <joint name="left_wrist_0" type="revolute">
    <origin rpy="0 0 0" xyz="0.199200513507 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="left_elbow"/>
    <child link="left_wrist_0"/>
    <limit effort="1" lower="-0.78539816339" upper="0.78539816339" velocity="1"/>
  </joint>
  <joint name="left_wrist_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="left_wrist_0"/>
    <child link="left_wrist"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="left_hand" type="fixed">
    <origin rpy="0 0 0" xyz="0.1 0 0"/>
    <parent link="left_wrist"/>
    <child link="left_hand"/>
  </joint>
  <link name="right_leg_mount"/>
  <link name="right_hip_0"/>
  <link name="right_hip_1"/>
  <link name="right_hip">
    <visual>
      <origin rpy="0 1.57079632679 0" xyz="0.233775716592 0 0"/>
      <geometry>
        <cylinder length="0.467551433183" radius="0.05"/>
      </geometry>
    </visual>
  </link>
  <link name="right_knee">
    <visual>
      <origin rpy="0 1.57079632679 0" xyz="0.148995297669 0 0"/>
      <geometry>
        <cylinder length="0.297990595338" radius="0.025"/>
      </geometry>
    </visual>
  </link>
  <link name="right_ankle_0"/>
  <link name="right_ankle">
    <visual>
      <origin rpy="0 0 0" xyz="0.05 0 0"/>
      <geometry>
        <box size="0.1 0.05 .005"/>
      </geometry>
    </visual>
  </link>
  <link name="right_foot"/>
  <!-- Joints -->
  <joint name="right_hip_leg_mount" type="fixed">
    <origin rpy="0 1.57079632679 0" xyz="0 -0.0729344655435 0"/>
    <parent link="base"/>
    <child link="right_leg_mount"/>
  </joint>
  <!-- hip : spherical -->
  <joint name="right_hip_0" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="right_leg_mount"/>
    <child link="right_hip_0"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="right_hip_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="right_hip_0"/>
    <child link="right_hip_1"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="right_hip_2" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="1 0 0"/>
    <parent link="right_hip_1"/>
    <child link="right_hip"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <!-- END hip -->
  <joint name="right_knee" type="revolute">
    <origin rpy="1.57079632679 0 0" xyz="0.467551433183 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="right_hip"/>
    <child link="right_knee"/>
    <limit effort="1" lower="-3.14159265359" upper="0.1" velocity="1"/>
  </joint>
  <!-- ankle : spherical -->
  <joint name="right_ankle_0" type="revolute">
    <origin rpy="-1.57079632679 0 1.57079632679" xyz="0.297990595338 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="right_knee"/>
    <child link="right_ankle_0"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="right_ankle_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="right_ankle_0"/>
    <child link="right_ankle"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <!-- END ankle -->
  <joint name="right_foot" type="fixed">
    <origin rpy="0 0 0" xyz="0.1 0 0"/>
    <parent link="right_ankle"/>
    <child link="right_foot"/>
  </joint>
  <link name="left_leg_mount"/>
  <link name="left_hip_0"/>
  <link name="left_hip_1"/>
  <link name="left_hip">
    <visual>
      <origin rpy="0 1.57079632679 0" xyz="0.233775716592 0 0"/>
      <geometry>
        <cylinder length="0.467551433183" radius="0.05"/>
      </geometry>
    </visual>
  </link>
  <link name="left_knee">
    <visual>
      <origin rpy="0 1.57079632679 0" xyz="0.148995297669 0 0"/>
      <geometry>
        <cylinder length="0.297990595338" radius="0.025"/>
      </geometry>
    </visual>
  </link>
  <link name="left_ankle_0"/>
  <link name="left_ankle">
    <visual>
      <origin rpy="0 0 0" xyz="0.05 0 0"/>
      <geometry>
        <box size="0.1 0.05 .005"/>
      </geometry>
    </visual>
  </link>
  <link name="left_foot"/>
  <!-- Joints -->
  <joint name="left_hip_leg_mount" type="fixed">
    <origin rpy="0 1.57079632679 0" xyz="0 0.0729344655435 0"/>
    <parent link="base"/>
    <child link="left_leg_mount"/>
  </joint>
  <!-- hip : spherical -->
  <joint name="left_hip_0" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="left_leg_mount"/>
    <child link="left_hip_0"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="left_hip_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="left_hip_0"/>
    <child link="left_hip_1"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="left_hip_2" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="1 0 0"/>
    <parent link="left_hip_1"/>
    <child link="left_hip"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <!-- END hip -->
  <joint name="left_knee" type="revolute">
    <origin rpy="1.57079632679 0 0" xyz="0.467551433183 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="left_hip"/>
    <child link="left_knee"/>
    <limit effort="1" lower="-3.14159265359" upper="0.1" velocity="1"/>
  </joint>
  <!-- ankle : spherical -->
  <joint name="left_ankle_0" type="revolute">
    <origin rpy="-1.57079632679 0 1.57079632679" xyz="0.297990595338 0 0"/>
    <axis xyz="0 0 1"/>
    <parent link="left_knee"/>
    <child link="left_ankle_0"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <joint name="left_ankle_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="left_ankle_0"/>
    <child link="left_ankle"/>
    <limit effort="1" lower="-1.57079632679" upper="1.57079632679" velocity="1"/>
  </joint>
  <!-- END ankle -->
  <joint name="left_foot" type="fixed">
    <origin rpy="0 0 0" xyz="0.1 0 0"/>
    <parent link="left_ankle"/>
    <child link="left_foot"/>
  </joint>
</robot>
